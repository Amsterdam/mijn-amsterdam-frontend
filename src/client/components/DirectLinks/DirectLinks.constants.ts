import { AppRoutes } from '../../../universal/config';
import { ExternalUrls } from '../../config/app';

export type DirectLinkType =
  | 'MIJN_SUBSIDIE'
  | 'MIJN_OVERHEID_BERICHTEBOX'
  | 'BUURT'
  | 'FAQ'
  | 'AMSTERDAM'
  | 'TIPS'
  | 'KREFIA';

export interface DirectLink {
  title: string;
  url?: string;
  isPhoneScreenLink?: true;
  isExternalLink?: true;
  id?: string;
}

export const LINKS: { [key in DirectLinkType]: DirectLink } = {
  BUURT: {
    title: 'Mijn buurt',
    url: AppRoutes.BUURT,
    isPhoneScreenLink: true,
    id: 'MyAreaHeader',
  },
  TIPS: {
    title: 'Mijn tips',
    url: AppRoutes.TIPS,
    isPhoneScreenLink: true,
    id: 'MyTipsHeader',
  },
  MIJN_SUBSIDIE: {
    title: 'Mijn Subsidies',
    url: ExternalUrls.MIJN_SUBSIDIES,
    isExternalLink: true,
  },
  KREFIA: {
    title: 'Kredieten en budgetbeheer (Krefia)',
    url: ExternalUrls.KREFIA,
    isExternalLink: true,
  },
  FAQ: {
    title: 'Veelgestelde vragen',
    url: ExternalUrls.MIJN_AMSTERDAM_VEELGEVRAAGD,
    isExternalLink: true,
  },
  MIJN_OVERHEID_BERICHTEBOX: {
    title: 'Mijn Overheid Berichtenbox',
    url: ExternalUrls.BERICHTENBOX,
    isExternalLink: true,
  },
  AMSTERDAM: {
    title: 'Amsterdam.nl',
    url: ExternalUrls.AMSTERDAM,
    isExternalLink: true,
  },
};
