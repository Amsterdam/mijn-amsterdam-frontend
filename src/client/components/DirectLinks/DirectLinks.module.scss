.DirectLinks {
  @include select(ams-heading) {
    margin-bottom: 1.6rem;
  }

  @include mq-phone() {
    margin-top: 2rem;
  }
}

.LinkList {
  @include reset-list-style;
  overflow: hidden; // Clearfix

  li {
    border-bottom: 0.1rem $color-neutral-grey2 solid;
    line-height: 5.5rem;
    float: left;
    width: calc(50% - 2.1rem);
  }

  li:nth-child(odd) {
    margin-right: 4rem;
  }

  li:nth-child(-n + 2) {
    border-top: 0.1rem $color-neutral-grey2 solid;
  }

  @include mq-phone() {
    // Reset the list style for smaller screens, just 1 column here
    li {
      &,
      &:nth-child(-n + 2),
      &:nth-child(odd) {
        width: auto;
        float: none;
        border-top: 0;
        margin-right: 0;
      }

      &:nth-child(1) {
        border-top: 0.1rem $color-neutral-grey2 solid;
      }
    }
  }

  li:nth-child(2n + 1) {
    clear: left;
  }

  a {
    margin: auto;
    font-weight: normal;

    svg {
      fill: var(--ams-link-color);
      height: 1.5rem;
      width: 1.5rem;
    }
  }

  svg {
    margin-right: 0.5rem;
  }
}
