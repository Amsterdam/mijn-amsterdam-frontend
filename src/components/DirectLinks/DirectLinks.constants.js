export const LINKS = {
  MIJN_BELASTINGEN: {
    displayName: 'Mijn belastingen',
    url: 'https://belastingbalie.amsterdam.nl',
  },
  // MIJN_PARKEREN: {
  //   displayName: 'Mijn parkeren',
  //   url: '',
  // },
  // MIJN_WERK_EN_INKOMEN: {
  //   displayName: 'Mijn werk & inkomen',
  //   url: '',
  // },
  MIJN_ERFPACHT: {
    displayName: 'Mijn erfpacht',
    url: 'https://mijnerfpacht.amsterdam.nl',
  },
};
