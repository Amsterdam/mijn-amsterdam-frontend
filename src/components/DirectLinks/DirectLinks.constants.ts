import { AppRoutes } from '../../App.constants';
export type DirectLinkType =
  | 'MIJN_BELASTINGEN'
  | 'MIJN_SUBSIDIE'
  | 'MIJN_WERK_EN_INKOMEN'
  | 'MY_AREA'
  | 'MY_TIPS'
  | 'MIJN_ERFPACHT';

export interface DirectLink {
  title: string;
  url: string;
  isSmallScreenLink?: true;
  isExternalLink?: true;
}

export const LINKS: { [key in DirectLinkType]: DirectLink } = {
  MY_TIPS: {
    title: 'Mijn tips',
    url: AppRoutes.MY_TIPS,
    isSmallScreenLink: true,
  },
  MY_AREA: {
    title: 'Mijn buurt',
    url: AppRoutes.MY_AREA,
    isSmallScreenLink: true,
  },
  MIJN_BELASTINGEN: {
    title: 'Mijn belastingen',
    url: 'https://belastingbalie.amsterdam.nl',
    isExternalLink: true,
  },
  MIJN_SUBSIDIE: {
    title: 'Mijn subsidie',
    url: 'https://mijnsubsidies.amsterdam.nl/loket/',
    isExternalLink: true,
  },
  MIJN_WERK_EN_INKOMEN: {
    title: 'Mijn werk & inkomen',
    url: 'https://edison.amsterdam.nl/SignIn?ReturnUrl=%2F',
    isExternalLink: true,
  },
  MIJN_ERFPACHT: {
    title: 'Mijn erfpacht',
    url: 'https://mijnerfpacht.amsterdam.nl',
    isExternalLink: true,
  },
};
