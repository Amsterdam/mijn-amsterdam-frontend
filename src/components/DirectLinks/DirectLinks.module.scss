@import 'styles/_global.scss';

.DirectLinks {
  width: $main-max-width;

  h2 {
    @include heading-large;
    margin-bottom: 1.6rem;
  }
}

.InnerContainer {
  width: $main-boxed-width;
  margin: 0 auto;
}

.LinkList {
  list-style-type: none;
  padding: 0;
  margin: 0;

  li {
    border-bottom: 0.1rem $color-neutral-grey2 solid;
    line-height: 5.5rem;
    float: left;
  }

  li:nth-child(odd) {
    width: calc(calc(#{$main-boxed-width} / 2) - 2rem);
    margin-right: 4rem;
  }

  li:nth-child(even) {
    width: calc(calc(#{$main-boxed-width} / 2) - 2rem);
  }

  li:nth-child(-n + 2) {
    border-top: 0.1rem $color-neutral-grey2 solid;
  }

  li:nth-child(2n + 1) {
    clear: left;
  }

  a {
    margin: auto;
    text-decoration: none;

    &:hover {
      color: $color-primary-red;
      text-decoration: underline;

      svg path {
        fill: $color-primary-red;
      }
    }
  }

  svg {
    margin-right: 0.5rem;
  }
}
