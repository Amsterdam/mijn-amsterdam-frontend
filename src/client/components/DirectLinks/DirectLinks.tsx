import { HTMLAttributes } from 'react';

import { Heading } from '@amsterdam/design-system-react';
import classnames from 'classnames';

import { LINKS } from './DirectLinks.constants';
import styles from './DirectLinks.module.scss';
import { entries } from '../../../universal/helpers/utils';
import { IconExternalLink } from '../../assets/icons';
import { ButtonBody } from '../Button/Button';
import { MaLink } from '../MaLink/MaLink';

export default function DirectLinks({
  id = 'DirectLinks',
  className,
  profileType,
  ...otherProps
}: HTMLAttributes<HTMLDivElement> & { profileType: ProfileType }) {
  return (
    <div
      {...otherProps}
      className={classnames(styles.DirectLinks, className)}
      id={id}
    >
      <Heading id={`${id}Header`} level={3} size="level-2">
        Bezoek ook
      </Heading>
      <ul className={styles.LinkList}>
        {entries(LINKS[profileType])
          .filter(([, { url, isActive }]) => !!url && isActive)
          .map((link) => {
            const [linkName, { url, title, isExternalLink, id }] = link;
            return (
              <li key={linkName}>
                <MaLink
                  id={id}
                  href={url}
                  isExternal={isExternalLink}
                  variant={'inline'}
                >
                  {isExternalLink ? (
                    <ButtonBody iconPosition="left" icon={IconExternalLink}>
                      {title}
                    </ButtonBody>
                  ) : (
                    title
                  )}
                </MaLink>
              </li>
            );
          })}
      </ul>
    </div>
  );
}
