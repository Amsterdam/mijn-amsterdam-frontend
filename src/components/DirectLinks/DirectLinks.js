import React from 'react';
import styles from './DirectLinks.module.scss';
import { LINKS } from './DirectLinks.constants';
import { ReactComponent as ExternalLinkIcon } from 'assets/icons/External-Link.svg';

export default function DirectLinks() {
  return (
    <div className={styles.DirectLinks}>
      <h2>Direct naar</h2>
      <ul className={styles.LinkList}>
        {Object.keys(LINKS).map(link => {
          const { url, title } = LINKS[link];
          return (
            <li key={link}>
              <a href={url}>
                <ExternalLinkIcon />
                {title}
              </a>
            </li>
          );
        })}
      </ul>
    </div>
  );
}
