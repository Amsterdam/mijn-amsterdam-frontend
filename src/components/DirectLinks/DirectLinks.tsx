import React from 'react';
import styles from './DirectLinks.module.scss';
import { LINKS } from './DirectLinks.constants';
import Heading from 'components/Heading/Heading';
import { ReactComponent as ExternalLinkIcon } from 'assets/icons/External-Link.svg';
import { entries } from 'helpers/App';
import { usePhoneScreen } from 'hooks/media.hook';
import ButtonLink from 'components/ButtonLink/ButtonLink';

export default function DirectLinks() {
  const isPhoneScreen = usePhoneScreen();
  return (
    <div className={styles.DirectLinks}>
      <Heading size="large">Bezoek ook</Heading>
      <ul className={styles.LinkList}>
        {entries(LINKS)
          .filter(([, { isPhoneScreenLink }]) =>
            !isPhoneScreen ? isPhoneScreenLink !== true : true
          )
          .map(link => {
            const [linkName, { url, title, isExternalLink }] = link;
            return (
              <li key={linkName}>
                {isExternalLink === true ? (
                  <a href={url}>
                    <ExternalLinkIcon />
                    {title}
                  </a>
                ) : (
                  <ButtonLink to={url}>{title}</ButtonLink>
                )}
              </li>
            );
          })}
      </ul>
    </div>
  );
}
