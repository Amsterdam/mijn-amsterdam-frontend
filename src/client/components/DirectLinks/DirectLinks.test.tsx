import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { usePhoneScreen } from '../../hooks/media.hook';
import DirectLinks from './DirectLinks';

vi.mock('../../hooks/media.hook');

describe('<DirectLinks />', () => {
  function getMountHtml(profileType: ProfileType) {
    return render(
      <BrowserRouter>
        <DirectLinks profileType={profileType} />
      </BrowserRouter>
    ).asFragment();
  }

  it('renders private links', () => {
    expect(getMountHtml('private')).toMatchSnapshot();
  });
  it('renders commercial links', () => {
    expect(getMountHtml('commercial')).toMatchSnapshot();
  });
  it('renders additional links on phone', () => {
    (usePhoneScreen as vi.Mock).mockReturnValueOnce(true);
    expect(getMountHtml('private')).toMatchSnapshot();
  });
});
