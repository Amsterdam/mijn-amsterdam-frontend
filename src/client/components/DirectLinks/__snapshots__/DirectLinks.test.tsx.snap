// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DirectLinks /> renders additional links on phone 1`] = `
<DocumentFragment>
  <div
    class="DirectLinks"
    id="DirectLinks"
  >
    <h3
      class="Heading Heading__large"
      id="DirectLinksHeader"
    >
      Bezoek ook
    </h3>
    <ul
      class="LinkList"
    />
  </div>
</DocumentFragment>
`;

exports[`<DirectLinks /> renders commercial links 1`] = `
<DocumentFragment>
  <div
    class="DirectLinks"
    id="DirectLinks"
  >
    <h3
      class="Heading Heading__large"
      id="DirectLinksHeader"
    >
      Bezoek ook
    </h3>
    <ul
      class="LinkList"
    >
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="/uitleg"
          id="Uitleg"
        >
          Dit ziet u in Mijn Amsterdam
        </a>
      </li>
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://www.amsterdam.nl/sbl/"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            External-Link.svg
          </svg>
          Stadsbank van Lening
        </a>
      </li>
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://www.amsterdam.nl/veelgevraagd/?productid={68422ECA-8C56-43EC-A9AA-B3DF190B5077}"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            External-Link.svg
          </svg>
          Veelgestelde vragen
        </a>
      </li>
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://www.digitaleoverheid.nl/dossiers/movo-mijn-overheid-voor-ondernemers/"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            External-Link.svg
          </svg>
          MijnOverheid voor ondernemers
        </a>
      </li>
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://www.amsterdam.nl"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            External-Link.svg
          </svg>
          Amsterdam.nl
        </a>
      </li>
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://www.amsterdam.nl/parkeren-verkeer/taxi/"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            External-Link.svg
          </svg>
          Taxiportaal
        </a>
      </li>
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://www.amsterdam.nl/parkeren-verkeer/varen-amsterdam/"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            External-Link.svg
          </svg>
          Varen in Amsterdam
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<DirectLinks /> renders private links 1`] = `
<DocumentFragment>
  <div
    class="DirectLinks"
    id="DirectLinks"
  >
    <h3
      class="Heading Heading__large"
      id="DirectLinksHeader"
    >
      Bezoek ook
    </h3>
    <ul
      class="LinkList"
    >
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://www.amsterdam.nl/sbl/"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            External-Link.svg
          </svg>
          Stadsbank van Lening
        </a>
      </li>
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://www.amsterdam.nl/veelgevraagd/?productid={68422ECA-8C56-43EC-A9AA-B3DF190B5077}"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            External-Link.svg
          </svg>
          Veelgestelde vragen
        </a>
      </li>
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://mijn.overheid.nl/berichtenbox/inbox/"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            External-Link.svg
          </svg>
          MijnOverheid Berichtenbox
        </a>
      </li>
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://www.amsterdam.nl"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            External-Link.svg
          </svg>
          Amsterdam.nl
        </a>
      </li>
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://www.amsterdam.nl/parkeren-verkeer/taxi/"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            External-Link.svg
          </svg>
          Taxiportaal
        </a>
      </li>
      <li>
        <a
          class="Button Button__plain Button__lean Linkd"
          href="https://www.amsterdam.nl/parkeren-verkeer/varen-amsterdam/"
          rel="external noopener noreferrer"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            External-Link.svg
          </svg>
          Varen in Amsterdam
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
